src/database.js

```diff
@@ -0,0 +1,25 @@
+const mongoose = require('mongoose');
+const PNIDSchema = require('./schema/pnid');
+const config = require('../config.json');
+
+const accountServerConfig = config.database.account;
+const accountServerURI = `mongodb://${accountServerConfig.address}:${accountServerConfig.port}/${accountServerConfig.database_name}`;
+let accountServerDBConnection;
+let PNID;
+
+async function connect() {
+	accountServerDBConnection = await mongoose.createConnection(accountServerURI, accountServerConfig.options);
+	accountServerDBConnection.on('error', console.error.bind(console, 'Mongoose connection error:'));
+	accountServerDBConnection.on('close', () => {
+		accountServerDBConnection.removeAllListeners();
+	});
+
+	PNID = accountServerDBConnection.model('PNID', PNIDSchema);
+
+	module.exports.PNID = PNID;
+}
```

Comment:
This doesn't wait until connected (but you do accept requests while not connected), use the `connect` event to resolve promise, use error event to reject.

Reply:
Or await the `connection.asPromise()`

Reply:
Fixed by awaiting `asPromise()`